tests/test_utils.py

```diff
@@ -94,9 +95,15 @@ def test_get_time_from_rfc3339_valid(rfc3339, seconds, tz):
     ('spam', '-----n', 10, '-', 'spam'),
     ('https://github.com/blah/spam.git', 'bacon', 10, '-', 'spam-bacon'),
 ])
-def test_make_name_from_git(repo, branch, limit, separator, expected):
-    assert make_name_from_git(repo, branch, limit, separator) == expected
+def test_make_name_from_git(repo, branch, limit, separator, expected, hash_size=5):
+    bc_name = make_name_from_git(repo, branch, limit + len(separator) + hash_size, separator, hash_size=hash_size)
 
+    assert re.match(re.escape(expected + separator) + (hash_size * '.'), bc_name)
```

Comment:
If we're doing this level of manipulation, we don't need regular expressions at all. Just trim off 'hash_size + 1' characters from the end before comparison.